tests/build-rss.test.js

```diff
@@ -0,0 +1,187 @@
+const fs = require('fs');
+const path = require('path');
+const rssFeed = require('../scripts/build-rss');
+const { mockRssData, title, type, desc, missingDateMockData, incompletePostMockData } = require('./fixtures/rssData');
+
+describe('rssFeed', () => {
+  const testOutputDir = path.join(__dirname, '..', 'public', 'test-output');
+  const outputPath = 'test-output/rss.xml';
+
+  beforeAll(() => {
+    if (!fs.existsSync(testOutputDir)) {
+      fs.mkdirSync(testOutputDir, { recursive: true });
+    }
+  });
+
+  afterAll(() => {
+    if (fs.existsSync(testOutputDir)) {
+      fs.readdirSync(testOutputDir).forEach(file => {
+        fs.unlinkSync(path.join(testOutputDir, file));
+      });
+      fs.rmdirSync(testOutputDir);
+    }
+  });
+
+  afterEach(() => {
+    jest.resetModules();
```

Comment:
@vishvamsinh28 any specific reason on why are we resetting all modules instead of mocks here?

Reply:
I tried using jest.resetAllMocks() and jest.clearAllMocks(), but they didn't work as expected because posts.json still contains cached data during the next test but jest.resetModules() clears everything loads with freshdata